pageserver/src/tenant/timeline.rs

```diff
@@ -1605,7 +1604,11 @@ impl Timeline {
                     );
                     let remote_layer = Arc::new(remote_layer);
 
-                    updates.insert_historic(remote_layer);
+                    if let Some(local_layer) = &local_layer {
+                        updates.replace_historic(local_layer, remote_layer)?;
+                    } else {
+                        updates.insert_historic(remote_layer)?;
+                    }
```

Comment:
We should only get here if the layer does not exist locally. We already printed the "remote layer does not exist locally, creating remote layer" message above too.

Reply:
Not exactly. We get there is layer doesn't exits locally OR it size is different from size of remote layer.
Before this PR just remove old layer in this case is insert new one.
Now I do not remove old layer explicitly but use replace.

Please notice that all fixes made in this PR are test-driven: without them some tests are failed.
In one case (import timeline from tar file) it was real bug. But in this case it is not actually a bug because 
we fist remove old layer. But as far as all this operations are buffered, this remove doesn't  immediately affect layer map.
It is certainly possible to locate pending deletions, but I prefered to use replace.